### PR TITLE
Travis: Make Selenium use Marionette instead of the add-on based driver

### PR DESCRIPTION
**Pull request by JohanLorenzo**
The current beta version of Firefox break the Selenium tests. This can be reproducible locally. It'll impact our tests once 47 reaches release.

From the Selenium server logs, the installation of the addon in Firefox is not supported anymore. The fix is to make Selenium plug into Marionette server.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "node-ssdp": "^2.7.0",
     "path": "^0.12.7",
     "raw-body": "^2.1.6",
-    "selenium-webdriver": "2.53.1",
+    "selenium-webdriver": "https://github.com/JohanLorenzo/selenium/releases/download/2.54.0-dev/selenium-webdriver-2.54.0-dev.tgz",
```
Comment (JohanLorenzo) [inline package.json:42]: I wish I could add a comment, but JSON doesn't allow it. This is a workaround until SeleniumHQ/selenium#2137 lands in master and a new version is released on npm (likely 2.54)

File: test/selenium/lib/setup_webapp.js [hunk 2]
```diff
@@ -4,14 +4,19 @@ var webdriver = require('selenium-webdriver');
...
 var SignInPageView = require('./view/sign_in/view.js');
 var MainView = require('./view/app_main/view.js');
+const Capabilities = require('selenium-webdriver/lib/capabilities').Capabilities;
 
-const driverBuilder = new webdriver.Builder().forBrowser('firefox');
+var capabilities = Capabilities.firefox();
+capabilities.set('marionette', true);
+const driverBuilder = new webdriver.Builder().withCapabilities(capabilities);
+
+console.log(driverBuilder.getCapabilities());
```
Comment (JohanLorenzo) [inline test/selenium/lib/setup_webapp.js]: Nit: Remove this line

**Comment by JohanLorenzo** [comment]
Build against [Firefox 47 is green](https://travis-ci.org/fxbox/foxbox/builds/132267978). So is [Firefox 46](https://travis-ci.org/fxbox/foxbox/builds/132269064) (that version supports both Marionette and the add-on driver).

r? @npark-mozilla

**Comment by nojunpark** [comment]
looks good to me, hopefully when this is merged, it's still working for me locally as well. r+

**Comment by nojunpark** [comment]
Actually, I rolled back my rust and ran this, and got following error on the first time I ran it, but the test is consistently passing after that, if this is caused by my setup, please ignore.

1) Test set up UI sessions ui once registred signin page should reject not matching passwords:
     WebDriverError: can't convert null to object
      at WebDriverError (node_modules/selenium-webdriver/lib/error.js:27:10)
      at Object.throwDecodedError (node_modules/selenium-webdriver/lib/error.js:611:11)
      at parseHttpResponse (node_modules/selenium-webdriver/http/index.js:518:15)
      at node_modules/selenium-webdriver/http/index.js:466:11
      at ManagedPromise.invokeCallback_ (node_modules/selenium-webdriver/lib/promise.js:1343:14)
      at TaskQueue.execute_ (node_modules/selenium-webdriver/lib/promise.js:2868:14)
      at TaskQueue.executeNext_ (node_modules/selenium-webdriver/lib/promise.js:2851:21)
      at node_modules/selenium-webdriver/lib/promise.js:2775:25
      at node_modules/selenium-webdriver/lib/promise.js:639:7
  From: Task: WebDriver.switchTo().alert()
      at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:386:17)
      at TargetLocator.alert (node_modules/selenium-webdriver/lib/webdriver.js:1591:29)
      at node_modules/selenium-webdriver/lib/until.js:117:30
      at node_modules/selenium-webdriver/lib/webdriver.js:718:14
      at TaskQueue.execute_ (node_modules/selenium-webdriver/lib/promise.js:2868:14)
      at TaskQueue.executeNext_ (node_modules/selenium-webdriver/lib/promise.js:2851:21)
      at node_modules/selenium-webdriver/lib/promise.js:2775:25
      at node_modules/selenium-webdriver/lib/promise.js:639:7
  From: Task: <anonymous>
      at pollCondition (node_modules/selenium-webdriver/lib/promise.js:2314:16)
      at node_modules/selenium-webdriver/lib/promise.js:2310:9
      at new ManagedPromise (node_modules/selenium-webdriver/lib/promise.js:1048:7)
      at node_modules/selenium-webdriver/lib/promise.js:2309:14
      at TaskQueue.execute_ (node_modules/selenium-webdriver/lib/promise.js:2868:14)
      at TaskQueue.executeNext_ (node_modules/selenium-webdriver/lib/promise.js:2851:21)
      at node_modules/selenium-webdriver/lib/promise.js:2775:25
      at node_modules/selenium-webdriver/lib/promise.js:639:7
  From: Task: Waiting for alert to be present
      at ControlFlow.wait (node_modules/selenium-webdriver/lib/promise.js:2307:17)
      at Driver.wait (node_modules/selenium-webdriver/lib/webdriver.js:714:29)
      at Object._waitForAlert (test/selenium/lib/view/alert.js:19:24)
      at Object.message (test/selenium/lib/view/alert.js:11:17)
      at test/selenium/lib/view/sign_in/view.js:25:39
      at ManagedPromise.invokeCallback_ (node_modules/selenium-webdriver/lib/promise.js:1343:14)
      at TaskQueue.execute_ (node_modules/selenium-webdriver/lib/promise.js:2868:14)
      at TaskQueue.executeNext_ (node_modules/selenium-webdriver/lib/promise.js:2851:21)
      at node_modules/selenium-webdriver/lib/promise.js:2730:27
      at node_modules/selenium-webdriver/lib/promise.js:639:7

**Comment by JohanLorenzo** [comment]
@npark-mozilla This error should have been worked around by [this selenium client](https://github.com/fxbox/foxbox/pull/489/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R41). Does the error occur after `npm install`? If so, I don't think I fixed the root cause of it :fearful: 

Edit: I just pushed an unbitrot revision. Nothing related to that particular failure

**Comment by nojunpark** [comment]
We could merge this, right?

EDIT: or right, r+.  I think above error was caused by me not running npm install beforehand.

**Pull request merged**